### PR TITLE
[core][dashboard] configurable timeouts

### PR DESCRIPTION
**Pull request by rynewang**
Ray Dashboard reorganizes stats of all nodes and all workers every 2s, and purges them every 600s. Make these 2 numbers configurable. Also removed 2 never used numbers.

### CODE REVIEW

File: python/ray/dashboard/consts.py [hunk 1]
```diff
@@ -22,10 +22,12 @@
 # Max allowed number of in-progress requests could be configured.
 RAY_STATE_SERVER_MAX_HTTP_REQUEST_ALLOWED = 1000
 
-RETRY_REDIS_CONNECTION_TIMES = 10
-CONNECT_REDIS_INTERNAL_SECONDS = 2
```
Comment (alexeykudinkin) [inline python/ray/dashboard/consts.py:26]: Are Redis configs dead code?

File: python/ray/dashboard/consts.py [hunk 2]
```diff
@@ -22,10 +22,12 @@
...
-PURGE_DATA_INTERVAL_SECONDS = 60 * 10
-ORGANIZE_DATA_INTERVAL_SECONDS = 2
+RAY_DASHBOARD_STATS_PURGING_INTERVAL = env_integer(
+    "RAY_DASHBOARD_STATS_PURGING_INTERVAL", 60 * 10
+)
+RAY_DASHBOARD_STATS_UPDATING_INTERVAL = env_integer(
+    "RAY_DASHBOARD_STATS_UPDATING_INTERVAL", 2
+)
```
Comment (alexeykudinkin) [inline python/ray/dashboard/consts.py:30]: Let's also make all updates in Node head configurable

File: python/ray/dashboard/consts.py (hunk 1 above)
Comment (rynewang) [reply python/ray/dashboard/consts.py:26]: dead, can't search out any usage

**Review by jjyao — APPROVED** [review]

**Comment by rynewang** [comment]
- GCS_CHECK_ALIVE_INTERVAL_SECONDS already configurable
- RAY_DASHBOARD_STATS_PURGING_INTERVAL, this PR
- RAY_DASHBOARD_STATS_UPDATING_INTERVAL, this PR
- EVENT_AGENT_REPORT_INTERVAL_SECONDS, this PR
- NODE_STATS_UPDATE_INTERVAL_SECONDS, this PR
- UPDATE_NODES_INTERVAL_SECONDS, this PR
- FREQUENTY_UPDATE_NODES_INTERVAL_SECONDS, this PR
- FREQUENT_UPDATE_TIMEOUT_SECONDS, this PR

File: python/ray/dashboard/consts.py (hunk 2 above)
Comment (rynewang) [reply python/ray/dashboard/consts.py:30]: done

**Review by alexeykudinkin — APPROVED** [review]

**Pull request merged**